galsim/chromatic.py

```diff
@@ -1567,9 +1661,26 @@ def drawImage(self, bandpass, image=None, integrator='trapezoidal', **kwargs):
             jac, offset, _ = self._getTransformations(bandpass.red_limit)
             int_im = galsim.Transform(int_im, jac=jac, offset=offset, gsparams=self.gsparams)
             image = int_im.drawImage(image=image, **kwargs)
+            self._last_wcs = image.wcs
             return image
         else:
-            return ChromaticObject.drawImage(self, bandpass, image, integrator, **kwargs)
+            image = ChromaticObject.drawImage(self, bandpass, image, integrator, **kwargs)
+            self._last_wcs = image.wcs
+            return image
+
+    @property
+    def noise(self):
```

Comment:
> I'm not sure we really need the bit of GSObject.__init__ that remains either (passing in an SBProfile

I think the logic there was to make sure that GSObjects always have an `SBProfile` attribute.  But given the way that Python works (unlike C++) it's just as easy to forget to call `GSObject.__init__(self, sbp)` as it is to set `self.SBProfile = sbp`.  So it's probably an unPythonic legacy of the fact that our initial python layer code was written by C++ coders.  :)